src/main/java/me/contaria/seedqueue/mixin/client/WorldGenerationProgressTrackerMixin.java

```diff
@@ -0,0 +1,105 @@
+package me.contaria.seedqueue.mixin.client;
+
+import it.unimi.dsi.fastutil.longs.Long2ObjectOpenHashMap;
+import me.contaria.seedqueue.SeedQueue;
+import me.contaria.seedqueue.interfaces.SQWorldGenerationProgressTracker;
+import net.minecraft.client.gui.WorldGenerationProgressTracker;
+import net.minecraft.server.WorldGenerationProgressLogger;
+import net.minecraft.util.Util;
+import net.minecraft.util.math.ChunkPos;
+import net.minecraft.world.chunk.ChunkStatus;
+import org.jetbrains.annotations.Nullable;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import java.util.Optional;
+
+@Mixin(WorldGenerationProgressTracker.class)
+public abstract class WorldGenerationProgressTrackerMixin implements SQWorldGenerationProgressTracker {
+    @Shadow @Final
+    private Long2ObjectOpenHashMap<ChunkStatus> chunkStatuses;
+    @Shadow
+    private ChunkPos spawnPos;
+    @Shadow @Final
+    private int radius;
+
+    @Shadow @Final private WorldGenerationProgressLogger progressLogger;
+    @Unique
+    private long freezeTime = -1;
+    @Unique
+    @Nullable
+    WorldGenerationProgressTracker frozenCopy = null;
+
+    @Unique
+    private int progressOverride = -1; // Overrides the returned progress percentage if equal to anything other than -1
+
+    @Inject(
+            method = "<init>",
+            at = @At("TAIL")
+    )
+    private void markFreezeTimeStart(int radius, CallbackInfo ci) {
+        long chunkMapFreezingTime = SeedQueue.config.chunkMapFreezing;
+        if (chunkMapFreezingTime != -1 && this.freezeTime == -1) {
+            this.makeFrozenCopyAfter(chunkMapFreezingTime);
+        }
+    }
+
+    @Inject(
+            method = "setChunkStatus",
+            at = @At(
+                    value = "INVOKE",
+                    target = "Lnet/minecraft/server/WorldGenerationProgressLogger;setChunkStatus(Lnet/minecraft/util/math/ChunkPos;Lnet/minecraft/world/chunk/ChunkStatus;)V",
+                    shift = At.Shift.BEFORE
```

Comment:
`Shift.BEFORE` is unnecessary, `@Inject`'s get put before the statement by default

Reply:
unnecessary ≠ bad
it is being explicit in where I want it to show up regardless of what sponge powered thinks the default should be

Reply:
According to people smarter than me using `Shift.BEFORE` results in the injection being put one instruction earlier
![grafik](https://github.com/user-attachments/assets/601efa7f-fa60-46d9-84d7-3151879a01dd)

Reply:
wtf